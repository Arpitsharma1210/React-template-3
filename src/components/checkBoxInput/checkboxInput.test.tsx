import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import CheckBoxInput from './';
import '@testing-library/jest-dom';

describe('CheckBox Input Component', () => {
  it('renders without crashing', () => {
    render(<CheckBoxInput />);
    expect(screen.getByRole('checkbox')).toBeInTheDocument();
  });

  it('renders label through props', () => {
    render(<CheckBoxInput label="I agree to terms" />);
    expect(screen.getByLabelText("I agree to terms")).toBeInTheDocument();
  });

  it('renders error message through props', () => {
    render(<CheckBoxInput error="An error occurred" />);
    expect(screen.getByText("An error occurred")).toBeInTheDocument();
  });

  it('runs onChange function passed through props', () => {
    const mockFunction = jest.fn();
    render(<CheckBoxInput onChange={mockFunction} />);
    fireEvent.click(screen.getByRole('checkbox'));
    expect(mockFunction).toHaveBeenCalled();
  });

  it('disables showing error through disableErrorMode prop', () => {
    render(<CheckBoxInput error="some error" disableErrorMode />);
    expect(screen.queryByText("some error")).toBeNull();
  });

  it('matches snapshot when given props', () => {
    const { asFragment } = render(
      <CheckBoxInput
        label="I agree to terms"
        error="An error occurred"
        value='false'
        fullWidth
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
