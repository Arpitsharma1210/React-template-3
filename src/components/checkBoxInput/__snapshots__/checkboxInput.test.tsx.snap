// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox Input Component matches snapshot when given props 1`] = `
<DocumentFragment>
  <div
    class="sc-bcHOTY jMBOyb"
  >
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd sc-iBIYEh cnZtCn css-j204z7-MuiFormControlLabel-root"
    >
      <span
        class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium sc-ftDPCE jnNHKQ css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
      >
        <input
          class="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate="false"
          type="checkbox"
          value="true"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckBoxOutlineBlankIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </span>
      <span
        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        I agree to terms
      </span>
    </label>
    <p
      class="MuiTypography-root MuiTypography-body2 sc-jTkTEj bXYaQ css-e784if-MuiTypography-root"
    >
      An error occurred
    </p>
  </div>
</DocumentFragment>
`;
